### ISSUE TITLE
implement `az acs list-locations`

### ISSUE THREAD

**Issue by bacongobbler** [Azure/azure-cli#2396]
Currently there is no way to figure out which data centers are able to create acs clusters. The only way to know right now as far as I know is to fail through trial-by-error, then use the regions listed in the error output. For example:

```
><> az acs create --resource-group="${AZURE_RG_NAME}" --location="canadaeast" --orchestrator-type=kubernetes --master-count=1 --agent-count=1 --agent-vm-size="Standard_D2_v2" --admin-username="k8sadmin" --name="${AZURE_SERVICE_NAME}" --dns-prefix="${AZURE_DNS_PREFIX}" --ssh-key-value @/home/bacongobbler/.ssh/id_rsa.pub 
waiting for AAD role to propagate.done
The provided location 'canadaeast' is not available for resource type 'Microsoft.ContainerService/containerServices'. List of available regions for the resource type is 'japaneast,centralus,eastus2,japanwest,eastasia,southcentralus,australiaeast,australiasoutheast,brazilsouth,southeastasia,westus,northcentralus,westeurope,northeurope,eastus'.
```

It would be great if there was a way to query which regions are available for the given resource type before creating the cluster. Right now it's a matter of picking one from `az account list-locations --query [].name --output tsv` and hoping it works.

**Comment by bacongobbler** [comment]
Also note that this later burned me with an obscure error which I eventually figured that my resource group was in `canadaeast` before creating the acs cluster: https://github.com/deis/workflow/issues/755

**Comment by rjtsdl** [comment]
This should be resolved by #4359. Please close this.

**Comment by itowlson** [comment]
Would be great to have this for AKS as well.
